cmd/a2l/a2l.go

```
@@ -68,114 +83,232 @@ func getTreeFromString(a2lString string) (result *a2l.RootNodeType, err error) {
 	emitUnpopulated := false
+	// Note: optionsParsed := false avoid to parse option for each chunk
+	optionsParsed := false
 
-	result = &a2l.JSONResponse{}
-
-	if request.Indent != nil {
-		for i := uint32(0); i < *request.Indent; i++ {
-			indent += " "
+	for {
+		request, err = stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				err = nil
+			}
+			break
 		}
-	}
-
-	if request.AllowPartial != nil {
-		allowPartial = *request.AllowPartial
-	}
 
-	if request.EmitUnpopulated != nil {
-		emitUnpopulated = *request.EmitUnpopulated
+		if !optionsParsed {
+			if request.Indent != nil {
+				for i := uint32(0); i < *request.Indent; i++ {
+					indent += " "
+				}
+			}
+			if request.AllowPartial != nil {
+				allowPartial = *request.AllowPartial
+			}
+			if request.EmitUnpopulated != nil {
+				emitUnpopulated = *request.EmitUnpopulated
+			}
+			optionsParsed = true
+		}
+		// Use a buffer to receive all chunks
+		buffer.Write(request.Tree)
 	}
-
-	opt := protojson.MarshalOptions{
-		AllowPartial:    allowPartial,
-		EmitUnpopulated: emitUnpopulated}
-
-	if rawData, parseError = opt.Marshal(request.Tree); parseError == nil {
-		// Note: see https://github.com/golang/protobuf/issues/1121
-		buffer := bytes.NewBuffer(indentedData)
-		if err = json.Indent(buffer, rawData, "", indent); err == nil {
-			result.Json = buffer.Bytes()
+	if err == nil {
+		if parseError = proto.Unmarshal(buffer.Bytes(), tree); parseError == nil {
+			opt := protojson.MarshalOptions{
+				AllowPartial:    allowPartial,
+				EmitUnpopulated: emitUnpopulated}
+
+			if rawData, parseError = opt.Marshal(tree); parseError == nil {
+				// Note: see https://github.com/golang/protobuf/issues/1121
+				var indentedBuffer bytes.Buffer
+				if err = json.Indent(&indentedBuffer, rawData, "", indent); err == nil {
+					rawData = indentedBuffer.Bytes()
+					for _, chunk = range chunkifyBySize(rawData, s.chunkSize) {
+						response.Json = chunk
+						err = stream.Send(response)
```

If the `stream.Send` function fails, the error is silently ignored here.